plugins/discovery-azure-classic/src/main/java/org/elasticsearch/plugin/discovery/azure/classic/AzureDiscoveryPlugin.java

```diff
@@ -88,36 +94,18 @@ public void onModule(DiscoveryModule discoveryModule) {
                             AzureComputeService.Discovery.ENDPOINT_NAME_SETTING);
     }
 
-    /**
-     * Check if discovery is meant to start
-     * @return true if we can start discovery features
-     */
-    private static boolean isDiscoveryReady(Settings settings, Logger logger) {
-        // User set discovery.type: azure
-        if (!AzureDiscoveryPlugin.AZURE.equalsIgnoreCase(DiscoveryModule.DISCOVERY_TYPE_SETTING.get(settings))) {
-            logger.trace("discovery.type not set to {}", AzureDiscoveryPlugin.AZURE);
-            return false;
-        }
-
-        if (isDefined(settings, AzureComputeService.Management.SUBSCRIPTION_ID_SETTING) &&
-            isDefined(settings, AzureComputeService.Management.SERVICE_NAME_SETTING) &&
-            isDefined(settings, AzureComputeService.Management.KEYSTORE_PATH_SETTING) &&
-            isDefined(settings, AzureComputeService.Management.KEYSTORE_PASSWORD_SETTING)) {
-            logger.trace("All required properties for Azure discovery are set!");
-            return true;
-        } else {
-            logger.debug("One or more Azure discovery settings are missing. " +
-                    "Check elasticsearch.yml file. Should have [{}], [{}], [{}] and [{}].",
-                AzureComputeService.Management.SUBSCRIPTION_ID_SETTING.getKey(),
-                AzureComputeService.Management.SERVICE_NAME_SETTING.getKey(),
-                AzureComputeService.Management.KEYSTORE_PATH_SETTING.getKey(),
-                AzureComputeService.Management.KEYSTORE_PASSWORD_SETTING.getKey());
-            return false;
+    @Override
+    public Settings additionalSettings() {
+        // For 5.0, the hosts provider was "zen", but this was before the discovery.zen.hosts_provider
+        // setting existed. This check looks for the legacy setting, and sets hosts provider if set
+        String discoveryType = DiscoveryModule.DISCOVERY_TYPE_SETTING.get(settings);
+        String hostsProvider = DiscoveryModule.DISCOVERY_HOSTS_PROVIDER_SETTING.get(settings);
+        if (hostsProvider == null && discoveryType.equals(AZURE)) {
```

Comment:
What do you mean by "not used"? All this does is check if the hosts provider setting has a value.